app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -625,7 +625,6 @@ public function resetPassword($email, $resetToken, $newPassword)
         $customerSecure->setRpTokenCreatedAt(null);
         $customerSecure->setPasswordHash($this->createPasswordHash($newPassword));
         $this->getAuthentication()->unlock($customer->getId());
-        $this->sessionManager->destroy();
         $this->destroyCustomerSessions($customer->getId());
```

Comment:
I don't think removing `destroy` is a right solution.

> Removed a code which were destroying session for current customer rather then for customer who requesting password reset

Can we destroy a proper session instead?